packages/app/src/serviceDashboard.ts

```diff
@@ -1,7 +1,37 @@
 import { TSource } from '@hyperdx/common-utils/dist/types';
 
-function getDefaults() {
+function getDefaults(jsonColumns: string[] = []) {
   const spanAttributeField = 'SpanAttributes';
+  const isJsonColumn = jsonColumns.includes(spanAttributeField);
+
+  const formatFieldAccess = (field: string, key: string) => {
+    if (isJsonColumn) {
+      return `${field}.\`${key}\``;
+    } else {
+      return `${field}['${key}']`;
+    }
+  };
+
+  let dbStatement = `coalesce(nullif(${formatFieldAccess(spanAttributeField, 'db.query.text')}, ''), nullif(${formatFieldAccess(spanAttributeField, 'db.statement')}, ''))`;
```

Comment:
Great catch on the outdated semantic convention! we can create a generic coalesce select generator so it can be reused for future semantic convention migration especially the custom dashboards